src/Field/CoordinateReferenceSystemCombo/CoordinateReferenceSystemCombo.jsx

```diff
@@ -0,0 +1,228 @@
+    /**
+     * An array of predefined crs definitions habving at least value (name of
+     * CRS) and code (e.g. EPSG-code of CRS) property
+     * @type {Array}
+     */
+    predefinedCrsDefinitions: PropTypes.arrayOf(PropTypes.shape({
+      value: PropTypes.string,
+      code: PropTypes.string
+    })),
+    /**
+     * A function
+     * @type {String}
+     */
+    onSelect: PropTypes.func
+  }
+
+  static defaultProps = {
+    emptyTextPlaceholderText: 'Please select a CRS',
+    epsgIoBaseUrl: 'https://epsg.io/',
+    onSelect: () => {}
+  }
+
+  /**
+   * Create a CRS combo.
+   * @constructs CoordinateReferenceSystemCombo
+   */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      crsDefinitions: [],
+      value: null
+    };
+
+    this.onCrsItemSelect = this.onCrsItemSelect.bind(this);
+  }
+
+  /**
+   * Fetch CRS definitions from epsg.io for given search string
+   *
+   * @param {String} searchVal The search string
+   * @param {Function} callback The callback function to be called on success.
+   */
+  fetchCrs = (searchVal, callback) => {
+    const { epsgIoBaseUrl } = this.props;
+
+    const queryParameters = {
+      format: 'json',
+      q: searchVal
+    };
+
+    fetch(`${epsgIoBaseUrl}?${UrlUtil.objectToRequestString(queryParameters)}`)
+      .then(response => response.json())
+      .then(this.onFetchSuccess.bind(this, callback))
+      .catch(this.onFetchError);
+  }
+
+  /**
+   * This function gets called on success of the EPSG.io fetch.
+   * It calls callback function if defined
+   *
+   * @param {Function} callback The callback function to be called
+   * @param {Object} json The result object of EPSG.io-API, see where
+   *                 https://github.com/klokantech/epsg.io#api-for-results
+   */
+  onFetchSuccess = (callback, json) => {
+    if (!callback) {
+      return false;
+    }
+    const results = json.results;
+    if (results && results.length > 0) {
+      // transform results
+      callback(results.map(obj => ({code: obj.code, value: obj.name, proj4def: obj.proj4, bbox: obj.bbox})));
+    } else {
+      callback([]);
+    }
+  }
+
+  /**
+   * This function gets called when the EPSG.io fetch returns an error.
+   * It logs the error to the console.
+   *
+   * @param {String} error The error string.
+   */
+  onFetchError(error) {
+    Logger.error(`Error while requesting in CoordinateReferenceSystemCombo: ${error}`);
+  }
+
+  /**
+   * This function gets called when the EPSG.io fetch returns an error.
+   * It logs the error to the console.
+   *
+   * @param {String} error The error string.
+   */
+  handleSearch = (value) => {
+    const {
+      predefinedCrsDefinitions
+    } = this.props;
+
+    if (!value || value.length === 0) {
+      this.setState({
+        value,
+        crsDefinitions: []
+      });
+      return;
+    }
+
+    if (!predefinedCrsDefinitions) {
+      this.fetchCrs(value, crsDefinitions => this.setState({ crsDefinitions, value })) ;
```

Comment:
The changes to `fetchCrs` would lead to further changes here... (see above)

```javascript
this.fetchCrs(value)
  .then(this.transformResults) // to be extracted from onFetchSucess
  .then(crsDefinitions => this.setState({ crsDefinitions, value })) // could be the new onFetchSucces
  .catch(Logger.error(`Error while requesting in CoordinateReferenceSystemCombo: ${error}`)); // could stay in onFetchError
```

Reply:
changed in 3cc756ffa0